libretro-rs/src/lib.rs

```diff
@@ -740,7 +723,14 @@ impl<T: RetroCore> RetroInstance<T> {
   /// Invoked by a `libretro` frontend, with the `retro_unload_game` API call.
   pub fn on_unload_game(&mut self) {
     let env = self.environment();
-    self.core_mut(|core| core.unload_game(&env))
+    self.core_mut(|core| core.unload_game(&env));
+    self.system = None;
+    self.audio_sample = None;
+    self.audio_sample_batch = None;
+    self.environment = None;
+    self.input_poll = None;
+    self.input_state = None;
+    self.video_refresh = None;
   }
```

Comment:
I think this is incorrect, a front-end could do these series of calls:

```
init
load_game
unload_game <-- callbacks removed
load_game <-- callbacks unavailable, likely will crash
unload_game
deinit
```

Simply unloading a game shouldn't cause all of the callbacks to disappear?

Reply:
Ah, I think you're right! I'll fix this tonight.

Reply:
I've had thoughts on how to prevent this sort of thing with state machines (#1), but alas I never got around to it.